podaac/swodlr_raster_create/submit_evaluate.py

```diff
@@ -0,0 +1,80 @@
+import json
+import logging
+from uuid import uuid4
+from .utils import (
+    mozart_client, get_param, search_datasets, load_json_schema
+)
+
+STAGE           = __name__.rsplit('.', 1)[1]
+DATASET_NAME    = 'SWOT_L2_HR_PIXCVec'
+PCM_RELEASE_TAG = get_param('sds_pcm_release_tag')
+
+validate_input  = load_json_schema('input')
+validate_jobset = load_json_schema('jobset')
+
+raster_eval_job_type = mozart_client.get_job_type(
+    f'job-SUBMIT_L2_HR_Raster:{PCM_RELEASE_TAG}'
+)
+raster_eval_job_type.initialize()
+
+def lambda_handler(event, _context):
+    logging.debug('Records received: %d', len(event['Records']))
+
+    jobs = [_process_record(record) for record in event['Records']]
+
+    job_set = {'jobs': jobs}
+    job_set = validate_jobset(job_set)
+    return job_set
+
+def _process_record(record):
```

Comment:
Just thinking through some of the failure scenarios, if the search fails or the job submit fails for some reason an exception will likely bubble up and cause the lambda to fail. Is that ok? I think the event would end up getting redelivered and retried so it may be alright. Alternatively we could think about some better error handling logic here.

Reply:
Error handling improved in latest commits to handle these issues more gracefully. The latest layout handles errors by passing them down the step function chain which eventually makes its way to `queue-update` to notify the rest of the system of the product's status. It includes information about the job id which the failure occurred (`job_id`), the microservice in which it occurred (`stage`), the traceback from the SDS if any (`traceback`), and a human readable list of error messages (`error`).

I was thinking of how we can possibly get the `bootstrap` lambda, which kickstarts the step function, to push failed jobs to a dead letter queue automatically once the step function is started; but it doesn't seem possible to do it on an exception in the step function from the current way we have it setup.